lib/Auth/Process/PerunIdentity.php

```diff
@@ -429,51 +429,49 @@ public static function unauthorized($request)
      */
     protected function getSPAttributes($spEntityID)
     {
+        $attrNames = [
+            $this->facilityCheckGroupMembershipAttr,
+            $this->facilityVoShortNamesAttr,
+            $this->facilityDynamicRegistrationAttr,
+            $this->facilityRegisterUrlAttr,
+            $this->facilityAllowRegistrationToGroupsAttr,
+        ];
+
         try {
             $facility = $this->rpcAdapter->getFacilityByEntityId($spEntityID);
 
             if ($facility === null) {
                 return;
             }
 
-            $checkGroupMembership = $this->rpcAdapter->getFacilityAttribute(
+            $facilityAttrValues = $this->rpcAdapter->getFacilityAttributesValues(
                 $facility,
-                $this->facilityCheckGroupMembershipAttr
+                $attrNames
             );
-            if ($checkGroupMembership !== null) {
-                $this->checkGroupMembership = $checkGroupMembership;
+
+            if (array_key_exists($this->facilityCheckGroupMembershipAttr, $facilityAttrValues)) {
+                $this->checkGroupMembership = $facilityAttrValues[$this->facilityCheckGroupMembershipAttr];
             }
 
-            $facilityVoShortNames = $this->rpcAdapter->getFacilityAttribute(
-                $facility,
-                $this->facilityVoShortNamesAttr
-            );
-            if (!empty($facilityVoShortNames)) {
-                $this->facilityVoShortNames = $facilityVoShortNames;
+            if (array_key_exists($this->facilityVoShortNamesAttr, $facilityAttrValues) &&
+                !empty($facilityAttrValues[$this->facilityVoShortNamesAttr])) {
+                $this->facilityVoShortNames = $facilityAttrValues[$this->facilityVoShortNamesAttr];
             }
 
-            $dynamicRegistration = $this->rpcAdapter->getFacilityAttribute(
-                $facility,
-                $this->facilityDynamicRegistrationAttr
-            );
-            if ($dynamicRegistration !== null) {
-                $this->dynamicRegistration = $dynamicRegistration;
+            if (array_key_exists($this->facilityDynamicRegistrationAttr, $facilityAttrValues)) {
+                $this->dynamicRegistration = $facilityAttrValues[$this->facilityDynamicRegistrationAttr];
             }
 
-            $this->registerUrl = $this->rpcAdapter->getFacilityAttribute(
-                $facility,
-                $this->facilityRegisterUrlAttr
-            );
-            if ($this->registerUrl === null) {
-                $this->registerUrl = $this->defaultRegisterUrl;
+            if (array_key_exists($this->facilityRegisterUrlAttr, $facilityAttrValues)) {
+                $this->registerUrl = $facilityAttrValues[$this->facilityRegisterUrlAttr];
+
+                if ($this->registerUrl === null) {
```

Comment:
This cannot be in the if block (lines: 465-470), because you want to set $this->registerUrl = $this->defaultRegisterUrl; it the array key not exist too.